### PR TITLE
Upgrade from Rails 4.2 to 5.0 and update gems [#177672607]

### PR DESCRIPTION
**Pull request by dougmartin**
Rails has been upgraded along with all the dependent gems but some code and gems (and all the geni* gems) have been commented out due to runtime errors.  I can load the homepage and login on my dev instance.

The tests are all breaking for this PR but there are a number of stories in PT that address the breakage so I want to merge this into the `rails5` tracking branch and start on those stories.

### CODE REVIEW

File: docs/rails-5-upgrade.md [hunk 1]
```diff
@@ -0,0 +1,136 @@
+# Rails 5 Upgrade Steps
+
+1. First attempt to update to last 5.0 version of rails (5.0.7.2)
+  1. Attempted upgrade by updating rails version in Gemfile and running `bundle update rails` inside a /bin/bash session in the app Docker container.  Got bundler error: `genigames_connector was resolved to 0.0.5, which depends on rails (~> 4.2)`
+2. Attempted upgrading genigames_connector from 4.2 to 5.0
+  1. Created `rails5-support` branch from `spike-rails4-support`
+  2. Changed `s.add_dependency "rails", "~> 4.2"` to `s.add_dependency "rails", "~> 5"` in gemspec file and updated version in gem to 0.0.6
+  3. Added `- ..:/projects` to volumes section of app in docker-compose.yml (this points to my root Concord projects folder)
+  4. Ran the following in a rigse Docker bash session: `bundle config local.genigames_connector /projects/genigames-connector/`
+  5. Changed rigse Gemfile to `gem 'genigames_connector', git: 'https://github.com/concord-consortium/genigames-connector.git', branch: 'rails5-support'`
```
Comment (knowuh) [inline docs/rails-5-upgrade.md:10]: clever 🧙

**Review by knowuh — APPROVED** [review]
Looks good.  Thanks!

File: docs/rails-5-upgrade.md (hunk 1 above)
Comment (dougmartin) [reply docs/rails-5-upgrade.md:10]: I knew there had to be a better way to develop local gems.  It works with the caveat that the gem has to have a .git folder - the version of bundler we are using doesn't support the newer `path:` option which works with a plain folder.

File: rails/config/application.rb [hunk 2]
```diff
@@ -185,7 +186,8 @@ class Application < Rails::Application
 
     # propagate errors normally just like in other Active Record callbacks
     # See https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#error-handling-in-transaction-callbacks
-    config.active_record.raise_in_transactional_callbacks = true
+    # RAILS 5 UPGRADE: deprecated
```
[the remark this answers is not in the document]
Comment (dougmartin) [reply rails/config/application.rb:189]: This was meant more as a review/posterity note and not something that needed to be addressed again.

**Pull request merged**